fatfs/go_fatfs.go

```diff
@@ -290,6 +290,34 @@ func (l *FATFS) OpenFile(path string, flags int) (tinyfs.File, error) {
 	return file, nil
 }
 
+// translateFlags translates osFlags such as os.O_RDONLY into fatfs flags.
+// http://elm-chan.org/fsw/ff/doc/open.html
+func translateFlags(osFlags int) C.BYTE {
+	var result C.BYTE
+	result = C.FA_READ
+	switch osFlags {
+	case os.O_RDONLY:
+		// r
+		result = C.FA_READ
+	case os.O_WRONLY | os.O_CREATE | os.O_TRUNC:
+		// w
+		result = C.FA_CREATE_ALWAYS | C.FA_WRITE
+	case os.O_WRONLY | os.O_CREATE | os.O_APPEND:
+		// a
+		result = C.FA_OPEN_APPEND | C.FA_WRITE
+	case os.O_RDWR:
+		// r+
+		result = C.FA_READ | C.FA_WRITE
+	case os.O_RDWR | os.O_CREATE | os.O_TRUNC:
+		// w+
+		result = C.FA_CREATE_ALWAYS | C.FA_WRITE | C.FA_READ
+	case os.O_RDWR | os.O_CREATE | os.O_APPEND:
+		// a+
```

Comment:
Should the `a+` case be assigned like this?

```go
result = C.FA_OPEN_APPEND | C.FA_WRITE | C.FA_READ
```

Reply:
Oops, I forgot to write it down.
I added it, squashed it, and rebased it.